File: 백준/Bronze/2920. 음계/음계.py
def ascending(num = list):
    for i in range(1,len(num)+1):
        if i != num[i-1]:
            return False
    return True

def descending(num = list):
    for i in range(len(num),0,-1):
        if i != num[len(num)-i]:
            return False
    return True       

import sys

num_list = list(map(int,sys.stdin.readline().split()))
if ascending(num_list):
    print('ascending')
elif descending(num_list):
    print('descending')
else:
    print('mixed')

File: 백준/Bronze/2920. 음계/음계.cc
#include <bits/stdc++.h>
using namespace std;

int main() 
{
	bool ascend = true;
	bool descend = true;
	vector<int> array;

	for (int i = 0; i < 8; ++i)
	{
		int temp;
		cin >> temp;
		array.push_back(temp);
	}

	for (int i = 1; i < 8; ++i)
	{
		if (array[i] > array[i - 1])
			descend = false;
		else if (array[i] < array[i - 1])
			ascend = false;
	}

	if (ascend)
		cout << "ascending";
	else if (descend)
		cout << "descending";
	else
		cout << "mixed";

	return 0;
}